index.js

```diff
@@ -0,0 +1,241 @@
+  return function (req, res, next) {
+    if (req.method !== 'OPTIONS' && module.exports.isProtectedRoute(req)) { // allow mocking of these for testing
+      module.exports.isAuthenticated(site)(req, res, next);
+    } else {
+      next();
+    }
+  };
+}
+
+/**
+ * middleware to show login page
+ * @param {function} tpl
+ * @param {object} site
+ * @param {array} currentProviders
+ * @returns {function}
+ */
+function onLogin(tpl, site, currentProviders) {
+  return function (req, res) {
+    const flash = req.flash();
+
+    if (flash && _includes(flash.error, 'Invalid username/password')) {
+      res.statusCode = 401;
+      res.setHeader('WWW-Authenticate', 'Basic realm="Incorrect Credentials"');
+      res.end('Access denied');
+      // this will prompt users one more time for the correct password,
+      // then display the login page WITHOUT saving the basic auth credentials.
+      // if they hit login, it'll show the auth form again, but won't show it a
+      // second time if they enter the wrong info.
+      // note: if the user enters the correct info on the second form,
+      // it'll show the login page but they'll have to click login again
+      // (and it'll automatically log them in without having to re-enter credentials)
+      // note: all of the above is the default behavior in amphora, but we're
+      // going to use varnish to automatically redirect them back to the ldap auth
+    } else {
+      res.send(tpl({
+        path: getPathOrBase(site),
+        flash: flash,
+        currentProviders: currentProviders,
+        user: req.user,
+        logoutLink: `${getAuthUrl(site)}/logout`
+      }));
+    }
+  };
+}
+
+/**
+ * middleware to log out. redirects to login page
+ * note: it goes to login page because usually users are navigating from edit mode,
+ * and they can't be redirected back into edit mode without logging in
+ * @param {object} site
+ * @returns {function}
+ */
+function onLogout(site) {
+  return function (req, res) {
+    req.logout();
+    res.redirect(`${getAuthUrl(site)}/login`);
+  };
+}
+
+/**
+ * There exists a case in which a user has an active session and
+ * is then removed as a user from a Clay instance. We must handle
+ * the error by re-directing the user to the login page and logging
+ * them out of their current session
+ *
+ * @param  {Object} site
+ * @returns {Function}
+ */
+function checkAuthentication(site) {
+  return function (err, req, res, next) {
+    if (err) {
+      onLogout(site)(req, res);
+    } else {
+      next();
+    }
+  };
+}
+
+/**
+ * initialize authentication
+ * @param {express.Router} router
+ * @param {array} providers (may be empty array)
+ * @param {object} site config for the site
+ * @param {object} storage
+ * @returns {array}
+ */
+function init(router, providers, site, storage) {
+  if (_isEmpty(providers)) {
+    return []; // exit early if no providers are passed in
+  }
+
+  setDb(storage);
+
+  const tpl = compileLoginPage('login.handlebars'),
```

Comment:
I see this method doesn't receive any parameters in the declaration, could you please take a look?